File: LeetCode/cpp/1877_Minimize_Maximum_Pair_Sum_in_Array.cpp
class Solution {
public:
	int minPairSum(vector<int>& nums) {
		int answer = 0;

		sort(nums.begin(), nums.end());

		for (int i = 0; i < nums.size() / 2; i++) {
			answer = max(answer, nums[i] + nums[nums.size() - 1 - i]);
		}

		return answer;
	}
};

File: LeetCode/java/1877_Minimize_Maximum_Pair_Sum_in_Array.java
import java.util.Arrays;

class Solution {
    public int minPairSum(int[] nums) {
        int answer = 0;

        Arrays.sort(nums);

        for (int i = 0; i < nums.length / 2; i++) {
            answer = Math.max(answer, nums[i] + nums[nums.length - 1 - i]);
        }

        return answer;
    }
}